src/Functions/Public/iaas/New-vRACloudAccountvSphere.ps1

```diff
@@ -243,7 +211,15 @@
                 $AssociatedCloudAccountIdsAddQuotes = $AssociatedCloudAccountIds | ForEach-Object { "`"$_`"" }
                 $AssociatedCloudAccountIdsFormatForBodyText = $AssociatedCloudAccountIdsAddQuotes -join ","
 
-                $RegionIDs = ProcessReqionIds # process to see if regions were given, if not, we pull all of them
+                if ($null -eq $RegionIds) {
+                    # process to see if regions were given, if not, we pull all of them
+                    if ($PSBoundParameters.ContainsKey("Credential")){
+                        $RegionIDs = Get-vRARegionEnumerationvSphere -HostName $HostName  -Credential $Credential -AcceptSelfSignedCertificate:$($AcceptSelfSignedCertificate)
+                    } else {
+                        # assume username and password
+                        $RegionIDs = Get-vRARegionEnumerationvSphere -HostName $HostName  -username $username -Password $Password -AcceptSelfSignedCertificate:$($AcceptSelfSignedCertificate)
```

Comment:
Variables should be PascalCase
`$username` -> `$Username`

Reply:
-username should be -Username

Reply:
Will be fixed next commit :)